r/lint.sh

```diff
@@ -17,11 +17,14 @@
 # specific language governing permissions and limitations
 # under the License.
 
+# This script requires Python 3 and clang-format, which should already be
+# on your system.
+
 SOURCE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 CPP_BUILD_SUPPORT=$SOURCE_DIR/../cpp/build-support
 
 # Run clang-format
-CLANG_FORMAT=clang-format-7
+CLANG_FORMAT=$(which clang-format)
```

Comment:
Hmm, it is very recommended to stick to `clang-format-7`, because different versions can produce different formatting (and then CI fails).

Reply:
The script failed for me locally because it said there was no such command, and I couldn't find any docs on how to obtain this or any other prerequisites for the script. But this worked and also worked on Travis on my fork. 

If it needs to be clang-format-7, can we either (1) provide instructions on how to get it, and/or (2) maybe do like

```
if [ "$CLANG_FORMAT" == "]; then
  CLANG_FORMAT=$(which clang-format)
fi
```

for a default, and then set that env var in the Travis job?

Reply:
It needs to be the same well-defined version of clang-format accross all scripts (CI, etc.). If we want to switch to version 8 or something, we'll need to do it everywhere.

I don't understand your (2) proposal. You would still have the problem that it does a different format than CI does.

Reply:
Then I need to know how to get `clang-format-7` such that I can run the script as is. Otherwise I won't run it and the build will fail (like this).

Reply:
What is your OS?

Reply:
I'm on macOS. `which clang-format` returns `/usr/local/bin/clang-format`; `which clang-format-7` returns nothing.

Reply:
Ah, I see. I cannot help you, perhaps there are some macOS users around. @xhochy perhaps?

Reply:
How about `: ${CLANG_FORMAT:=clang-format-7}` for local run on macOS?
We can run `CLANG_FORMAT=$(which clang-format) ./lint.sh` on macOS with the same `lint.sh`.

BTW, if you install LLVM by Homebrew, `CLANG_FORMAT=$(brew --prefix llvm@7)/bin/clang-format ./lint.sh` is better.

Reply:
Thanks @kou, I'll make that change and document the recommendation